Source/WebCore/dom/TextEncoderStreamEncoder.cpp

```diff
@@ -44,22 +45,21 @@ RefPtr<Uint8Array> TextEncoderStreamEncoder::encode(const String& input)
     for (size_t cptr = 0; cptr < view.length(); cptr++) {
         // https://encoding.spec.whatwg.org/#convert-code-unit-to-scalar-value
         auto token = view[cptr];
-        if (m_pendingHighSurrogate) {
-            auto highSurrogate = *m_pendingHighSurrogate;
-            m_pendingHighSurrogate = std::nullopt;
-            if (token >= 0xDC00 && token <= 0xDFFF) {
-                auto codePoint = 0x10000 + ((highSurrogate - 0xD800) << 10) + (token - 0xDC00);
+        if (m_pendingLeadSurrogate) {
+            auto leadSurrogate = *std::exchange(m_pendingLeadSurrogate, std::nullopt);
+            if (U_IS_TRAIL(token)) {
```

Comment:
I think this is a code unit and not a code point and so should be `U16_IS_TRAIL`. In practice either macro will work.